### PR TITLE
Worked on porting tensorflow to version 1.1. All unit test passes

### PR DESCRIPTION
**Pull request by ethantang95**
majority of the changes are within the tensorflow work or tool files. All the unit tests passes.

Some commits were done with respect to IDE files but they are all removed.

### CODE REVIEW

File: digits/dataset/images/classification/test_views.py [hunk 1]
```diff
@@ -117,13 +117,15 @@ def create_dataset(cls, **kwargs):
...
         }
+
         data.update(kwargs)
 
         request_json = data.pop('json', False)
         url = '/datasets/images/classification'
         if request_json:
             url += '.json'
 
+        print("Sending request to: " + url)
```
Comment (gheinrich) [inline digits/dataset/images/classification/test_views.py]: can you clean these up please?

**Review by gheinrich — COMMENTED** [review]
Thanks @ethantang95 can you remove unnecessary changes please?

**Comment by ethantang95** [comment]
done

File: .gitignore [hunk 2]
```diff
@@ -17,3 +17,10 @@ TAGS
 /build/
 /dist/
 *.egg-info/
+
+# intellij files
+.idea/workspace.xml
+.idea
+
+# vscode
+.vscode
```
Comment (gheinrich) [inline .gitignore]: files should end with a blank line

File: digits/config/tensorflow.py [hunk 3]
```diff
@@ -34,7 +34,7 @@ def test_tf_import(python_exe):
 
 if not tf_enabled:
     print('Tensorflow support disabled.')
-#    print('Failed importing Tensorflow with python executable "%s"\n%s' % (tf_python_exe, err))
+    print('Failed importing Tensorflow with python executable "%s"\n%s' % (tf_python_exe, err))
```
Comment (gheinrich) [inline digits/config/tensorflow.py]: doesn't seem necessary

File: digits/dataset/images/classification/views.py [hunk 4]
```diff
@@ -310,9 +310,11 @@ def create():
         )
 
         if form.method.data == 'folder':
+            print("going into folder")
```
Comment (gheinrich) [inline digits/dataset/images/classification/views.py]: please remove *all* print statements

File: digits/job.py [hunk 5]
```diff
@@ -13,11 +13,14 @@
 from .status import Status, StatusCls
 from digits.config import config_value
 from digits.utils import sizeof_fmt, filesystem as fs
+import logging
 
 # NOTE: Increment this everytime the pickled object changes
 PICKLE_VERSION = 2
 
 
+logger = logging.getLogger('digits.tools.inference')
```
Comment (gheinrich) [inline digits/job.py]: doesn't seem necessary

File: digits/model/images/classification/test_views.py [hunk 6]
```diff
@@ -288,7 +288,9 @@ class BaseViewsTestWithModel(BaseViewsTestWithDataset):
     def setUpClass(cls):
         super(BaseViewsTestWithModel, cls).setUpClass()
         cls.model_id = cls.create_model(json=True)
-        assert cls.model_wait_completion(cls.model_id) == 'Done', 'create failed'
+        result = cls.model_wait_completion(cls.model_id)
```
Comment (gheinrich) [inline digits/model/images/classification/test_views.py]: doesn't seem necessary

File: digits/test_views.py [hunk 7]
```diff
@@ -144,6 +144,8 @@ def job_wait_completion(cls, job_id, timeout=10, polling_period=0.5, job_type='j
                     job_id, url, rv.status_code)
                 assert job_id in rv.data
                 return status
+            if (time.time() - start) > timeout:
+                print("Job " + job_id + " took too long to finish")
```
Comment (gheinrich) [inline digits/test_views.py]: unnecessary

File: digits/tools/inference.py [hunk 8]
```diff
@@ -63,15 +65,15 @@ def infer(input_list,
         task.dataset = dataset
 
     # retrieve snapshot file
-    task = model.train_task()
+    trained_task = model.train_task()
```
Comment (gheinrich) [inline digits/tools/inference.py]: unnecessary

File: digits/tools/tensorflow/model.py [hunk 9]
```diff
@@ -54,13 +54,13 @@ def average_gradients(tower_grads):
                 # Append on a 'tower' dimension which we will average over below.
                 grads.append(expanded_g)
             # Average over the 'tower' dimension.
-            grad = tf.concat(0, grads)
-            grad = tf.reduce_mean(grad, 0)
+            grads_transformed = tf.concat(grads, 0)
+            grads_transformed = tf.reduce_mean(grads_transformed, 0)
```
Comment (gheinrich) [inline digits/tools/tensorflow/model.py:58]: are you sure about that?

File: docs/DevelopmentSetup.md [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+
+## Running DIGITS in Development
+
+DIGITS comes with the script to run for a development server.
+To run the development server, use
+```
+./digits-devserver
+```
+
+Running the development server allow you to <TBD>
```
Comment (gheinrich) [inline docs/DevelopmentSetup.md]: either remove or complete sentence

**Review by gheinrich — CHANGES_REQUESTED** [review]
Hi @ethantang95 can you remove unnecessary changes, squash your commits and update https://github.com/gheinrich/DIGITS-GAN/blob/master/scripts/travis/install-tensorflow.sh to install the right version of Tensorflow.

File: digits/tools/tensorflow/model.py (hunk 9 above)
Comment (ethantang95) [reply digits/tools/tensorflow/model.py:58]: Yeah, the new API swaps the place of them

File: digits/job.py [hunk 11]
```diff
@@ -13,11 +13,11 @@
 from .status import Status, StatusCls
 from digits.config import config_value
 from digits.utils import sizeof_fmt, filesystem as fs
+import logging
```
Comment (gheinrich) [inline digits/job.py]: seems unnecessary

File: digits/model/images/classification/test_views.py [hunk 12]
```diff
@@ -661,6 +661,7 @@ def test_classify_one(self):
             # StringIO wrapping is needed to simulate POST file upload.
             image_upload = (StringIO(infile.read()), 'image.png')
 
+        print("task id="+str(self.model_id))
```
Comment (gheinrich) [inline digits/model/images/classification/test_views.py]: this too

File: digits/model/job.py [hunk 13]
```diff
@@ -5,9 +5,12 @@
 from digits.job import Job
 from digits.utils import override
 
+import logging
+
```
Comment (gheinrich) [inline digits/model/job.py]: this too

File: digits/tools/inference.py [hunk 14]
```diff
@@ -50,6 +51,7 @@ def infer(input_list,
     if jobs_dir == 'none':
         jobs_dir = digits.config.config_value('jobs_dir')
 
+
```
Comment (gheinrich) [inline digits/tools/inference.py]: this too

File: digits/tools/inference.py [hunk 15]
```diff
@@ -24,6 +24,7 @@
 
 # Import digits.config before caffe to set the path
 import caffe_pb2  # noqa
+import json
```
Comment (gheinrich) [inline digits/tools/inference.py]: we don't need this

File: digits/tools/tensorflow/model.py (hunk 9 above)
Comment (gheinrich) [reply digits/tools/tensorflow/model.py:58]: ah OK thanks can you make sure multi-GPU training still works?

**Review by gheinrich — COMMENTED** [review]
can you also squash your commits and fix `./digits-lint` errors?

File: digits/tools/tensorflow/model.py (hunk 9 above)
Comment (ethantang95) [reply digits/tools/tensorflow/model.py:58]: I do not have a multi-gpu system, know who I can talk to for one?

**Comment by ethantang95** [comment]
all changes done

**Review by gheinrich — APPROVED** [review]

**Pull request merged**